src/erlcloud_s3.erl

```diff
@@ -1200,11 +1200,17 @@ s3_result_fun(#aws_request{response_type = ok} = Request) ->
 s3_result_fun(#aws_request{response_type = error,
                            error_type = aws,
                            response_status = Status} = Request) when
+%% Retry conflicting operations 409,Conflict and 500s.
+      Status =:= 409; Status >= 500 ->
+    Request#aws_request{should_retry = true};
+s3_result_fun(#aws_request{response_type = error,
+                           error_type = aws,
+                           response_status = Status} = Request) when
 %% Retry for 400, Bad Request is needed due to Amazon 
 %% returns it in case of throttling.
-%% Also retry conflictin operations 409,Conflict.
-      Status =:= 400; Status =:= 409; Status >= 500 ->
-    Request#aws_request{should_retry = true};
+      Status =:= 400 ->
+    ShouldRetry = erlcloud_aws:is_throttling_error_response(Request),
```

Comment:
Is `s3` API throttled at all? I don't see anything mentioning it here: http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html